### PR TITLE
Add GC9A01 Driver

### PR DESCRIPTION
**Pull request by lmatte7**
This PR adds driver support for the Waveshare 1.28in circular [display](https://www.waveshare.com/1.28inch-lcd-module.htm). A lot of the code was based on the driver code for ST7735 with specific updates for this board. The full datasheet is available [here](https://www.waveshare.com/w/upload/5/5e/GC9A01A.pdf).

The driver has been tested and can successfully run the display with a RPI Pico and a ESP32.

### CODE REVIEW

**Comment by deadprogram** [comment]
Any comments @sago35 or @conejoninja before merging?

File: gc9a01/gc9a01.go
Comment (sago35) [inline gc9a01/gc9a01.go]: Make it to satisfy drivers.Displayer.

https://github.com/tinygo-org/drivers/blob/v0.22.0/displayer.go#L5-L14

File: gc9a01/gc9a01.go
Comment (sago35) [inline gc9a01/gc9a01.go]: Make it to satisfy drivers.Displayer.

https://github.com/tinygo-org/drivers/blob/v0.22.0/displayer.go#L5-L14

File: gc9a01/gc9a01.go
Comment (sago35) [inline gc9a01/gc9a01.go]: Make it to satisfy drivers.Displayer.

https://github.com/tinygo-org/drivers/blob/v0.22.0/displayer.go#L5-L14

**Review by sago35 — CHANGES_REQUESTED** [review]
@lmatte7 
For the most part, I think it is very good.
I commented because I would like to make the same design as other drivers.

Please update README.md.
Add a test to smoke-test in the Makefile.

If possible, please keep the interface in the same form as the st7735 and ili9341 interfaces.
In other words, we want int16 to be used instead of int32 in many places.
https://github.com/tinygo-org/drivers/blob/release/st7735/st7735.go
https://github.com/tinygo-org/drivers/blob/v0.22.0/ili9341/ili9341.go

**Comment by lmatte7** [comment]
@sago35 Thanks for the feedback! I’ll make the changes and send an update.

**Comment by sago35** [comment]
I will review it tomorrow or so.

File: examples/gc9a01/main.go
Comment (sago35) [inline examples/gc9a01/main.go:34]: Please add sleep, etc. so that the main function does not terminate.
https://github.com/tinygo-org/drivers/blob/release/examples/ili9341/basic/main.go#L33-L35

**Review by sago35 — CHANGES_REQUESTED** [review]
@lmatte7 
Please add sleep to examples.
I will approve if you add it.

**Review by sago35 — APPROVED** [review]
LGTM
Thank you!

**Comment by deadprogram** [comment]
Now going to squash/merge thank you for working on this @lmatte7

**Pull request merged**
